plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -742,23 +742,31 @@ public static Set<HivePrivilegeInfo> parsePrivilege(PrivilegeGrantInfo userGrant
 
     public static HiveBasicStatistics getHiveBasicStatistics(Map<String, String> parameters)
     {
-        OptionalLong numFiles = parse(parameters.get(NUM_FILES));
-        OptionalLong numRows = parse(parameters.get(NUM_ROWS));
-        OptionalLong inMemoryDataSizeInBytes = parse(parameters.get(RAW_DATA_SIZE));
-        OptionalLong onDiskDataSizeInBytes = parse(parameters.get(TOTAL_SIZE));
-        return new HiveBasicStatistics(numFiles, numRows, inMemoryDataSizeInBytes, onDiskDataSizeInBytes);
+        return getHiveBasicStatistics(parameters, false);
     }
 
-    private static OptionalLong parse(@Nullable String parameterValue)
+    public static HiveBasicStatistics getHiveBasicStatistics(Map<String, String> parameters, boolean useSparkFallback)
     {
-        if (parameterValue == null) {
-            return OptionalLong.empty();
-        }
-        Long longValue = Longs.tryParse(parameterValue);
-        if (longValue == null || longValue < 0) {
-            return OptionalLong.empty();
+        OptionalLong numFiles = toLong(parameters.get(NUM_FILES));
+        OptionalLong numRows = toLong(parameters.get(NUM_ROWS));
+        OptionalLong inMemoryDataSizeInBytes = toLong(parameters.get(RAW_DATA_SIZE));
+        OptionalLong onDiskDataSizeInBytes = toLong(parameters.get(TOTAL_SIZE));
+
+        if (useSparkFallback) {
+            if ((numFiles.isEmpty() || numFiles.getAsLong() == 0) && parameters.containsKey(ThriftSparkMetastoreUtil.STATS_PREFIX + NUM_FILES)) {
+                numFiles = toLong(parameters.get(ThriftSparkMetastoreUtil.STATS_PREFIX + NUM_FILES));
+            }
+            if ((numRows.isEmpty() || numRows.getAsLong() == 0) && parameters.containsKey(ThriftSparkMetastoreUtil.STATS_PREFIX + NUM_ROWS)) {
```

Comment:
We don't need to do fallbacks at individual parameter level. The main important parameter here is row count, so we can change `if (useSparkFallback)` -> `if (useSparkStatistics && numRows.isEmpty())` and remove the checks for individual parameters.
Also, zero statistic is a valid statistic if it exists, so please remove the `getAsLong() == 0` checks